.github/workflows/ci_tests_legacy.yaml

```diff
@@ -95,10 +78,18 @@ jobs:
         env:
           GH_TOKEN: ${{ github.token }}
 
+      # Install uv package manager
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+
       # Install the package that we want to test
       - name: Install the package
-        run: make install
+        run: |
+          uv run --with pip==23 --resolution lowest-direct --all-extras --dev make install
+          uv pip list
 
       # Run the tests but skip images
       - name: Run tests
-        run: make test_no_images PYTEST_EXTRA="-r P"
+        run: uv run --with pytest==8 --resolution lowest-direct --all-extras --dev make test_no_images PYTEST_EXTRA="-r P"
+        env:
+          GMT_LIBRARY_PATH: $CONDA_PREFIX/lib
```

Comment:
Getting this error at https://github.com/GenericMappingTools/pygmt/actions/runs/11945150103/job/33297317606?pr=3639#step:7:35:

```python-traceback
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/runner/work/pygmt/pygmt/pygmt/__init__.py", line 24, in <module>
    from pygmt import datasets
  File "/home/runner/work/pygmt/pygmt/pygmt/datasets/__init__.py", line 7, in <module>
    from pygmt.datasets.earth_age import load_earth_age
  File "/home/runner/work/pygmt/pygmt/pygmt/datasets/earth_age.py", line 12, in <module>
    from pygmt.datasets.load_remote_dataset import _load_remote_dataset
  File "/home/runner/work/pygmt/pygmt/pygmt/datasets/load_remote_dataset.py", line 9, in <module>
    from pygmt.clib import Session
  File "/home/runner/work/pygmt/pygmt/pygmt/clib/__init__.py", line 9, in <module>
    from pygmt.clib.session import Session, __gmt_version__
  File "/home/runner/work/pygmt/pygmt/pygmt/clib/session.py", line 111, in <module>
    _libgmt = load_libgmt()
  File "/home/runner/work/pygmt/pygmt/pygmt/clib/loading.py", line 62, in load_libgmt
    raise GMTCLibNotFoundError("\n".join(error_msg))
pygmt.exceptions.GMTCLibNotFoundError: Error loading GMT shared library at '/home/runner/micromamba/envs/pygmt/lib/libgmt.so'.
/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /home/runner/micromamba/envs/pygmt/lib/./libgdal.so.34)
Error loading GMT shared library at 'libgmt.so'.
/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /home/runner/micromamba/envs/pygmt/lib/python3.10/lib-dynload/../.././libgdal.so.34)
make: *** [Makefile:35: _runtest] Error 1
``` 

Similar to issues mentioned at https://github.com/GenericMappingTools/try-gmt/issues/50#issue-1781117291 and https://github.com/GenericMappingTools/try-gmt/pull/37#issuecomment-1367074465.